rawkit/raw.py

```diff
@@ -175,6 +175,57 @@ def save_thumb(self, filename=None):
         self.libraw.libraw_dcraw_thumb_writer(
             self.data, filename.encode('ascii'))
 
+    def get_4_col_raw(self):
+        """
+        Read the 4 colour raw data
+        The returned numpy array is (given colour RGGB)
+        ---------
+        R B R B ... 
+        G B G B ...
+        R B R B ....
+        . . . . .
+        . . . .  .
+        . . . .   .
+        --------
+
+
+        Returns:
+            array: 4 colour data of the image in unit16. (width x height)
+            str : colour channel description (ie. RGGB, RGBG)
+        """
+        # Unpack the data, so that rawdata is populated
+        self.unpack()
+        rawdata = self.data.contents.rawdata
+
+        # Return None if the image isn't 4 colour, which will happen for some
+        # cameras
+
+        try:
+            rawdata.color4_image.contents
+        except:
```

Comment:
Ah, this makes sense, thanks! Let's catch the actual exception here though; bare excepts can have unintended side effects :)